File: in_Java/0067 Add Binary.java
class Solution {
    public String addBinary(String a, String b) {
        char[] ch_a = a.toCharArray();
        char[] ch_b = b.toCharArray();
        StringBuilder sb = new StringBuilder();
        int i = a.length()-1;
        int j = b.length()-1;
        int over = 0;
        int cur;
        int x;
        int y;
        while(i >= 0 || j >= 0){
            x = (i < 0)? 0 : Character.getNumericValue(ch_a[i]);
            y = (j < 0)? 0 : Character.getNumericValue(ch_b[j]);
            cur = x + y + over;
            over = cur / 2;
            cur = cur % 2;
            sb.insert(0,String.valueOf(cur));
            i--;
            j--;
        }
        String res = sb.toString();
        if(over == 1) return "1" + res;
        else return res;
    }
}

File: in_Python/0067 Add Binary.py
class Solution(object):
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        
        i = len(a)-1
        j = len(b)-1
        c = 0
        res = ""
        while i >= 0 or j >= 0:
            if i >= 0:
                x = int(a[i])
            else:
                x = 0
            if j >= 0:
                y = int(b[j])
            else:
                y = 0
            z = x + y + c
            res = str(z % 2) + res
            c = z / 2
            i -= 1
            j -= 1
        if c == 1:
            res = '1' + res
        return res
        